agir/front/components/toolsPage/ToolsPage.js

```diff
@@ -0,0 +1,578 @@
+
+    if (childrenSize <= containerSize) {
+      setShowLeftScroll(false);
+      setShowRightScroll(false);
+      return;
+    }
+
+    if (scrollLeft <= 0) {
+      setShowLeftScroll(false);
+      setShowRightScroll(true);
+      return;
+    }
+    if (scrollLeft + containerSize >= childrenSize) {
+      setShowLeftScroll(true);
+      setShowRightScroll(false);
+      return;
+    }
+
+    if (scrollLeft > 0) setShowLeftScroll(true);
+    if (scrollLeft + containerSize < childrenSize) setShowRightScroll(true);
+  };
+
+  const handleScroll = (direction) => {
+    const containerSize = containerRef.current?.offsetWidth;
+    let position = containerRef.current.scrollLeft - containerSize;
+    if (direction == "right")
+      position = containerRef.current.scrollLeft + containerSize;
+    containerRef.current.scrollTo({
+      top: 0,
+      left: position,
+      behavior: "smooth",
+    });
+    setUpdateCarrousel(true);
+  };
+
+  const CarrouselArrow = ({ direction }) => (
+    <CarrouselArrowContainer
+      toRight={direction === "right"}
+      onClick={() => handleScroll(direction)}
+    >
+      <RawFeatherIcon name="chevron-right" />
+    </CarrouselArrowContainer>
+  );
+
+  useEffect(() => {
+    let timeout;
+    if (isUpdateCarrousel) {
+      timeout = setTimeout(() => {
+        updateCarrousel();
+        setUpdateCarrousel(false);
+      }, 600);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [pages, isUpdateCarrousel, containerRef]);
+
+  if (!pages) return null;
+
+  return (
+    <ListItemActionContainer>
+      {showLeftScroll && <CarrouselArrow direction="left" />}
+
+      <Carrousel ref={containerRef}>
+        {pages?.map((page, id) => (
+          <ItemAction
+            key={id}
+            image={page.img}
+            title={page.title}
+            href={page.link}
+          />
+        ))}
+      </Carrousel>
+
+      {showRightScroll && <CarrouselArrow direction="right" />}
+    </ListItemActionContainer>
+  );
+};
+ListItemAction.propTypes = {
+  pages: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
+      title: PropTypes.string,
+      img: PropTypes.string,
+      link: PropTypes.string,
+      category_id: PropTypes.number,
+      category_name: PropTypes.string,
+    })
+  ),
+};
+
+const ItemWebsite = ({ img, href, title }) => (
+  <Link href={href}>
+    <ItemWebsiteContainer img={img}>
+      <div />
+      <div>
+        <div>{title}</div>
+        <div>
+          <RawFeatherIcon
+            name="arrow-right"
+            color={style.black1000}
+            width="1.25rem"
+          />
+        </div>
+      </div>
+    </ItemWebsiteContainer>
+  </Link>
+);
+ItemWebsite.propTypes = {
+  title: PropTypes.string.isRequired,
+  href: PropTypes.string.isRequired,
+  img: PropTypes.string.isRequired,
+};
+
+const ToolsPage = () => {
+  const isDesktop = useIsDesktop();
+  const [categories, pages] = getWPPagesAndCategories();
```

Comment:
Il faut qu'on sache que c'est un hook React custom : 
```suggestion
  const [categories, pages] = useWPPagesAndCategories();
```